### PR TITLE
[backplane-2.2]refactor rbacfinalizercontroller to fix cluster ns terminating

### PR DESCRIPTION
**Pull request by zhiweiyin318**
fix issue: https://issues.redhat.com/browse/ACM-5789

### CODE REVIEW

File: pkg/hub/manager.go [hunk 1]
```diff
@@ -119,10 +119,10 @@ func RunControllerManager(ctx context.Context, controllerContext *controllercmd.
 	)
 
 	rbacFinalizerController := rbacfinalizerdeletion.NewFinalizeController(
-		kubeInfomers.Rbac().V1().Roles(),
-		kubeInfomers.Rbac().V1().RoleBindings(),
-		kubeInfomers.Core().V1().Namespaces().Lister(),
-		clusterInformers.Cluster().V1().ManagedClusters().Lister(),
+		kubeInfomers.Rbac().V1().Roles().Lister(),
+		kubeInfomers.Rbac().V1().RoleBindings().Lister(),
+		kubeInfomers.Core().V1().Namespaces(),
```
Comment (skeeey) [inline pkg/hub/manager.go]: we seem not filter the cluster label in the  kube informer in the 2.2, can do we add the cluster label filter in the kube inform now?

File: pkg/hub/rbacfinalizerdeletion/controller.go [hunk 2]
```diff
@@ -44,76 +43,78 @@ type finalizeController struct {
...
 			key, _ := cache.MetaNamespaceKeyFunc(obj)
 			return key
-		}, roleInformer.Informer(), roleBindingInformer.Informer()).
+		}, clusterInformer.Informer()).
+		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			accessor, _ := meta.Accessor(obj)
+			if accessor.GetDeletionTimestamp().IsZero() {
+				return ""
+			}
+			if _, ok := accessor.GetLabels()[clusterLabelKey]; !ok {
```
Comment (skeeey) [inline pkg/hub/rbacfinalizerdeletion/controller.go]: if we filter the cluster label in the informer, we do not need this

File: pkg/hub/manager.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/hub/manager.go]: added

File: pkg/hub/rbacfinalizerdeletion/controller.go (hunk 2 above)
Comment (zhiweiyin318) [reply pkg/hub/rbacfinalizerdeletion/controller.go]: done

**Pull request merged**